core/src/data_sources/data_source.rs

```diff
@@ -404,21 +436,36 @@ impl DataSource {
                 &remove_tags,
             )
             .await?;
+        self.upsert_field_to_document_payload(qdrant_client, document_id, "tags", new_tags.clone())
```

Comment:
are you running cargo fmt on your code?

Reply:
I have a VS code extenstion that regularly formats the code and decided to let this line as is. I had never used cargo fmt before, but I just did and it stayed silent on this (and on the whole code). How should this have been formatted?

Reply:
maybe I ran it wrong?

Reply:
It feels quite long and some other parts of your code look like not indented exactly well.

Do you honour `.rustfmt.toml`?

Reply:
the toml I have has maxWidth=100 and the line is exactly 100 chars, explains why we feel it's long
rustfmt does not complain either